src/control/XournalMain.cpp

```diff
@@ -382,7 +382,7 @@ auto XournalMain::run(int argc, char* argv[]) -> int {
 
         try {
             if (fs::exists(p)) {
-                opened = control->openFile(p, openAtPageNumber);
+                opened = control->openFile(p, openAtPageNumber - 1);  // First page for user is page 1
```

Comment:
What happens if you use zero as input?
Make sure this doesn't fail. Or if it fails, that it will fail verbose.

if `openAtPageNumber` is a signed type, you could use something like `std::max(0, openAtPageNumber - 1)` if it's a unsigned type `std::max(1, openAtPageNumber) - 1` could do the job.

Reply:
If zero (or any negative integer) is used as input, `Control::fileLoaded` will eventually load the document opened at the same page as it was loaded last time (without any complaints) if the metadata is valid. If the metadata is invalid, it will open on the first page.

Maybe I could initialize `openAtPageNumber` by 0 instead of -1, since -1 is subtracted now. But it does not really make any difference, since -1 and -2 are both negative.